src/rcheevos/operand.c

```diff
@@ -64,6 +65,54 @@ static int rc_parse_operand_lua(rc_operand_t* self, const char** memaddr, rc_par
   return RC_OK;
 }
 
+static int rc_parse_operand_variable(rc_operand_t* self, const char** memaddr/*, rc_parse_state_t* parse*/) {
+  const char* aux = *memaddr;
+  size_t i;
+  char varName[RC_VALUE_MAX_NAME_LENGTH + 1] = { 0 };
+
+  for (i = 0; i < RC_VALUE_MAX_NAME_LENGTH && *aux != '}'; i++) {
+    switch (*aux) {
+
+    default:
+      if (rc_is_valid_variable_character(*aux, i == 0)) {
+        varName[i] = *aux;
+        /* strncat_s(varName, RC_VALUE_MAX_NAME_LENGTH + 1, aux, 1); */
+        ++aux;
+      }
+      else {
+        ++aux;
+        return RC_INVALID_VARIABLE_NAME;
+      }
+      break;
+
+    case '\0':/* end of string */
+    case '_': /* next condition */
+    case 'S': /* next condset */
+    case ')': /* end of macro */
+    case '$': /* maximum of values */
+      return RC_INVALID_VARIABLE_OPERAND;
```

Comment:
None of these should matter if we're expecting the variable name to be surrounded by brackets. Either it'll be a valid character (part of the name) or it won't.